plugin/pkg/admission/imagepolicy/admission.go

```diff
@@ -181,6 +181,20 @@ func (a *Plugin) admitPod(pod *api.Pod, attributes admission.Attributes, review
 		return errors.New("one or more images rejected by webhook backend")
 	}
 
+	if review.Status.Annotations == nil {
+		return nil
+	}
+	reviewAnnotations := a.filterAnnotations(review.Status.Annotations)
+	if len(reviewAnnotations) > 0 {
+		annotations := pod.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		for k, v := range reviewAnnotations {
+			annotations[k] = v
```

Comment:
We're OK with this overwriting existing annotations?

Reply:
I'd argue yes.

One example:  Customer specifies break-glass annotation, the system can then overwrite to indicate it was not needed.

I think not overwriting has the larger abuse potential (e.g., deployer sets a value to prevent a "this should be audited" annotation).

Reply:
> I think not overwriting has the larger abuse potential (e.g., deployer sets a value to prevent a "this should be audited" annotation).

can't an annotation be overwritten/changed later? trusting the presence and value of an annotation on a pod seems like a fragile thing to depend on for auditing or as a way to transfer security-related data between components

Reply:
yes, but this version should go into the log without the user having access to modify it in the meantime IIUC.  And I'd hope any later modification would also be in the audit logs.

But agreed that none of this is a security barrier, real auditing will re-do the checks and report just based on the deployed images.  I meant this more in the "accidental abuse" than in the "malicious abuse" sense that gives some benefit until the more complete auditing solution is available.